cmd/sshpiperd/daemon.go

```diff
@@ -217,19 +218,32 @@ func (d *daemon) run() error {
 			var uphook func([]byte) ([]byte, error)
 			var downhook func([]byte) ([]byte, error)
 
-			if d.recorddir != "" {
-				recorddir := path.Join(d.recorddir, p.DownstreamConnMeta().User())
+			uniqID := plugin.GetUniqueID(p.ChallengeContext())
+			if d.asciicastdir != "" {
+				recorddir := path.Join(d.asciicastdir, uniqID)
 				err = os.MkdirAll(recorddir, 0700)
 				if err != nil {
 					log.Errorf("cannot create screen recording dir %v: %v", recorddir, err)
 					return
 				}
+				recorder, err := newAsciicastLogger(recorddir)
+				if err != nil {
+					log.Errorf("cannot create screen recording logger: %v", err)
+					return
+				}
+				defer recorder.Close()
 
+				uphook = recorder.uphook
+				downhook = recorder.downhook
+			} else if d.recorddir != "" {
+				recorddir := path.Join(d.recorddir, uniqID)
```

Comment:
could you please add the feature flag to use old path format, this can be by default to new format
it would allow time for users to transit to new format